compiler/src/iree/compiler/Codegen/Common/ReconcileTranslationInfo.cpp

```diff
@@ -362,77 +364,112 @@ void ReconcileTranslationInfoPass::runOnOperation() {
   auto variantOp = getOperation();
   auto innerModuleOp = variantOp.getInnerModule();
 
-  auto exportOps = variantOp.getOps<IREE::HAL::ExecutableExportOp>();
+  // Get the symbol table of the inner module to lookup exported functions.
+  SymbolTable symbolTable(innerModuleOp);
+
+  // Construct the call-graph for the inner module. We traverse this when
+  // reconciling translation info.
+  CallGraph callGraph(innerModuleOp);
 
-  // reconciliation for multiple export ops is unsupported.
-  if (!llvm::hasSingleElement(exportOps)) {
-    return;
-  }
-  auto exportOp = *exportOps.begin();
   IRRewriter rewriter(&getContext());
+  auto exportOps = variantOp.getOps<IREE::HAL::ExecutableExportOp>();
 
-  SmallVector<IREE::Codegen::TranslationInfoAttr> translationInfos;
-  auto walkResult =
-      innerModuleOp->walk([&](FunctionOpInterface funcOp) -> WalkResult {
-        // Resolve workgroup distribution related `scf.forall` ops.
-        if (failed(resolveWorkgroupForAll(rewriter, funcOp, distributeAlong))) {
-          return failure();
-        }
+  for (auto exportOp : exportOps) {
+    SmallVector<IREE::Codegen::TranslationInfoAttr> translationInfos;
+    auto rootFuncOp = llvm::dyn_cast_if_present<FunctionOpInterface>(
+        symbolTable.lookup(exportOp.getSymNameAttr()));
+    if (!rootFuncOp || rootFuncOp.isExternal()) {
+      // Skip external functions.
+      continue;
+    }
+    // Resolve workgroup distribution related `scf.forall` ops. Today only a
```

Comment:
I think this comment here is not necessary. The limitation of what is supported in terms of `scf.forall` resolution is part of the `resolveWorkgoupForAll` function.